nservicebus/transactional-session/index.md

```diff
@@ -17,17 +17,18 @@ Consider an ASP.NET Core controller that creates a user in the business database
     * This results is a user in the database, known as a phantom record, which is never announced to the rest of the system.
 2. **Ghost message**: The controller publishes the `UserCreated` event first, then creates the user in the database. If a failure occurs between these two operations:
     * The `UserCreated` event is published, but the user is not created in the database.
-    * The rest of the system is notified about the creation of the user, but the record does not exist in the database. This causes further errors in the system which expects the user to exist in the database.
+    * The rest of the system is notified about the creation of the user, but the record does not exist in the database. This inconsistency causes errors in the system, as parts of the system expect the user to exist in the database.
 
-In the context of a message handler, the [Outbox](/nservicebus/outbox) feature can mitigate this problem, however, such scenarios remain unsolved outside of the context of a message handler.
+In the context of a message handler, the [Outbox](/nservicebus/outbox) feature can mitigate this problem. However, such scenarios remain unsolved outside the context of a message handler.
 
 A common technique to address this problem on the client side is to defer all operations to a message handler. This entails sending a message to create the user and publishing the `UserCreated`-event from within a message handler.
 However, there are scenarios where this approach is not feasible:
-- Existing applications that want to introduce messaging already have quite some logic in controllers. Moving all that logic into dedicated message handlers requires a lot of effort, and might no be feasible from day one.
-- Existing logic in the controller might assume certain side-effects to occur within the scope of the request (for example validation, notifications or error-handling) and not yet ready to fully embrace the asynchronous and fire&forget nature of offloading the work into message handlers.
-- There may be other scenarios in which it's not feasible to delay the database operation.
 
-The `TransactionalSession` feature solves this problem exactly.
+* Existing applications that want to introduce messaging already have quite some logic in controllers. Moving all that logic into dedicated message handlers requires significant effort, and might not be feasible from day one.
+* Existing logic in the controller may assume specific side effects to occur within the scope of the request, e.g., validation, notifications, or error handling. The logic may, therefore, not be ready to fully embrace the asynchronous and fire&forget nature of offloading work into message handlers.
+* There may be other scenarios in which it's not feasible to delay the database operation.
+
+The `TransactionalSession` feature solves this problem for messages sent/published outside the context of a message handler when combined with the [Outbox](/nservicebus/outbox) feature.
```

Comment:
```suggestion
The `TransactionalSession`, when combined with [Outbox](/nservicebus/outbox), solves this problem for messages sent and/or published outside of the context of a message handler.
```

Reply:
According to my grammar check "of the context of" is wrong. I will slightly tweak the suggestion